terasoluna-gfw-common-libraries/terasoluna-gfw-validator/src/test/java/org/terasoluna/gfw/common/validator/constraints/ByteMaxTest.java

```diff
@@ -143,6 +154,102 @@ public void testAnnotateUnexpectedType() throws Throwable {
         validator.validate(form, UnexpectedType.class);
     }
 
+    /**
+     * all values in the collection are valid.
+     * @throws Throwable
+     */
+    @Test
+    public void testCollectionValid() throws Throwable {
+        form.setListProperty(Arrays.asList("ああ", "ああ"));
+        Validator validator = Validation.buildDefaultValidatorFactory()
+                .getValidator();
+        Set<ConstraintViolation<ByteMaxTestForm>> violations = validator
+                .validate(form);
+
+        assertThat(violations, is(notNullValue()));
+        assertThat(violations.size(), is(0));
+    }
+
+    /**
+     * first value in the collection is invalid.
+     * @throws Throwable
+     */
+    @Test
+    public void testCollectionFirstInvalid() throws Throwable {
+        form.setListProperty(Arrays.asList("ああa", "ああ"));
+        Validator validator = Validation.buildDefaultValidatorFactory()
+                .getValidator();
+        Set<ConstraintViolation<ByteMaxTestForm>> violations = validator
+                .validate(form);
+
+        assertThat(violations, is(notNullValue()));
+        assertThat(violations.size(), is(1));
+        assertThat(violations.iterator().next().getPropertyPath().toString(),
+                is("listProperty[0].<list element>"));
+        assertThat(violations.iterator().next().getMessage(), is(String.format(
```

Comment:
Why create iterator twice...?

Reply:
5f389dab72c526edde87468d566f5dea034b52
Not need create iterator twice .
I fixed above commit .